### PR TITLE
Allow maintainers to have admin rights if requested

### LINKED ISSUE #616: Align repository maintainer/admin access
Our repository guidelines at https://github.com/open-telemetry/community/blob/6a33400d608709c8a9de6ffaf1bbcc117f622692/docs/how-to-configure-new-repository.md say:

> 5. The team `foo-maintainers` has `Maintain` permissions for the repository.
> 6. ...
> 7. Some repositories may include more individuals with `Admin` permissions.
   Typically to help set up repository, CI, web hooks or other administrative work.

According to the guideline above, only some individuals, if any, should be made an Admin.
The status quo, however, is that some OTel repos assign their maintainers team the Admin role, whereas others only give them Maintain permissions. We should either switch all to Maintain (+ individual Admins, if needed) or update the guidelines to formally allow this.

See https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization for a description of the roles. Admin is required to change the default branch (#402), for example.

### LINKED ISSUE #1232: REQUEST: Repository maintenance on opentelemetry-go-build-tools
Apply permission rules for maintainers https://github.com/open-telemetry/community/blob/4e4f113d64c3de8ce26b924b6c74867110c91aef/docs/how-to-configure-new-repository.md#permissions

This repo has maintainers as Admins which do not follow the rules.

### PR DESCRIPTION
**Pull request by trask**
Unblocks #1205 and #982

Resolves #1232
Resolves #616

As ~discussed~ voted on in the Governance Committee meeting 12/1/2022.

We will revisit this in 1 year, in case we are at a different place maturity-wise, and possibly tooling-wise, to remove Admin rights without causing burden on maintainers.

### CODE REVIEW

**Review by dyladan — APPROVED** [review]

**Comment by dyladan** [comment]
I think we should have in this repo the same file but which documents all the default settings a repo is expected to have. That way you can only document the deviations in your repository and link to here for anyone that wants to know the other settings but doesn't have permission to view them.

File: docs/default-repository-settings.md
Comment (trask) [inline docs/default-repository-settings.md:71]: #1259 can expand on this

File: docs/default-repository-settings.md
Comment (trask) [inline docs/default-repository-settings.md:36]: this section was moved over from `how-to-configure-new-repository.md`

I removed the image because it was outdated and didn't have anything that wasn't in the text

File: docs/default-repository-settings.md
Comment (trask) [inline docs/default-repository-settings.md:59]: this one wasn't in the prior text, but it was in the screenshot, so I included it but noted below that it was optional

File: docs/how-to-configure-new-repository.md
Comment (trask) [inline docs/how-to-configure-new-repository.md:61]: I removed this image because it was very outdated and "Require branches to be up to date before merging" was the only thing that wasn't already in the text, so I added that to the text (as optional with default true)

**Review by jpkrohling — APPROVED** [review]

File: docs/how-to-configure-new-repository.md
Comment (yurishkuro) [inline docs/how-to-configure-new-repository.md:19]: Does "child" mean a subset? Could we say a subset?

File: docs/how-to-configure-new-repository.md
Comment (yurishkuro) [inline docs/how-to-configure-new-repository.md:33]: I am not sure that adding a new file is beneficial here. The previous section is called "Permissions", yet doesn't mention the actual permissions and does not refer to the new file. And conceptually there's no difference between "how to configure new repo" and "default repo settings" to warrant a 2nd file.

File: docs/how-to-configure-new-repository.md
Comment (trask) [reply docs/how-to-configure-new-repository.md:19]: github calls them [child teams](https://docs.github.com/en/organizations/organizing-members-into-teams/requesting-to-add-a-child-team), though I always have to think twice about which is the parent and which is the child

maybe instead something like?

```suggestion
   and `foo-maintainers`. `foo-maintainers` team is a member of the `foo-approvers` team, and
```

File: docs/how-to-configure-new-repository.md
Comment (trask) [reply docs/how-to-configure-new-repository.md:33]: the second file was requested by @dyladan https://github.com/open-telemetry/community/pull/1311#issuecomment-1335733993

> I think we should have in this repo the same file but which documents all the default settings a repo is expected to have. That way you can only document the deviations in your repository and link to here for anyone that wants to know the other settings but doesn't have permission to view them.
>
> edit: it would also help with setting up repositories and auditing the settings of existing repositories.

I would defer to you and other @open-telemetry/technical-committee folks for whatever is most convenient since you are setting up the repos.

Comment (trask) [reply docs/how-to-configure-new-repository.md:33]: @yurishkuro I removed the new file in a different branch to see what it looks like, is this what you were imagining:

https://github.com/trask/community/blob/admin-rights-inlined/docs/how-to-configure-new-repository.md

Comment (yurishkuro) [reply docs/how-to-configure-new-repository.md:33]: @trask yep, looks great in one file, thanks!

File: docs/how-to-configure-new-repository.md
Comment (yurishkuro) [inline docs/how-to-configure-new-repository.md:30]: > as it each group always contains

typo?

**Review by yurishkuro — APPROVED** [review]

File: docs/how-to-configure-new-repository.md
Comment (trask) [reply docs/how-to-configure-new-repository.md]: I re-wrote this section a bit to make it clearer

**Pull request merged**
